File: src/main/java/com/pubnub/api/models/server/PresenceEnvelope.java
package com.pubnub.api.models.server;

import com.fasterxml.jackson.databind.JsonNode;
import lombok.Getter;

@Getter
public class PresenceEnvelope {

    private String action;
    private String uuid;
    private Integer occupancy;
    private Long timestamp;
    private JsonNode data;

}

File: src/main/kotlin/com/pubnub/api/models/server/PresenceEnvelope.kt
package com.pubnub.api.models.server

import com.google.gson.JsonElement

class PresenceEnvelope(
    internal val action: String? = null,
    internal val uuid: String? = null,
    internal val occupancy: Int? = null,
    internal val timestamp: Long? = null,
    internal val data: JsonElement? = null
)
